u4-databases/features/uni-controller.js

```diff
@@ -0,0 +1,54 @@
+const { db } = require('../_services/firebase-initialized')
+
+const iterationLists = async (req, res) => {
+  const query = db.collection('iterations').get()
+  const iteration = (await query).docs.map(doc => doc.data())
+
+  res.render('uni-iteration-list', { iteration })
+}
+
+const iterationDetails = async (req, res) => {
+  const code = req.params.code
+  const query = db.collection('iterations').doc(code).get()
+  const iteration = (await query).data()
+
+  res.render('uni-iteration-details', { iteration })
+}
+
+const studentDetails = async (req, res) => {
+  const code = req.params.code
+  const query = db.collection('students').doc(code).get()
+  const student = (await query).data()
+
+  res.render('uni-student-details', { student })
+}
+
+const courseLists = async (req, res) => {
+  const query = db.collection('iterations')
+    .where('year', '==', 2021)
+    .where('semester', '==', 1)
+    .get()
+  const iteration = (await query).docs.map(doc => doc.data())
+
+  res.render('uni-course-list', { iteration })
```

Comment:
คุณอาจตั้งชื่อตัวแปรนี้เพื่อแสดงว่าข้อมูลถูกกรองแล้ว เช่น `activeIterations`. หรือเนื่องจากเราใช้ข้อมูลการวนซ้ำเพื่อค้นหาหลักสูตรที่ใช้งาน คุณจึงสามารถใช้ `courses` ได้

You might name this variable to show that the data was filtered, ex: `activeIterations`. Or since we're using the iterations data to find active courses, you could use `courses`.